### ISSUE TITLE
Messages from ReactDevTools backend are recorded with some delay

### ISSUE THREAD

**Issue by hbenl** [replayio/gecko-dev#349]
This recording shows the issue:
http://localhost:8080/view?id=9cad4dcd-3b25-4473-9665-4da7a0af862f

**Comment by bhackett1024** [comment]
Can you try using console.log() at the calls to `__RECORD_REPLAY_REACT_DEVTOOLS_SEND_BRIDGE__` (https://github.com/RecordReplay/gecko-dev/blob/5c85e5c35ce23cf0debaac26542ce82d43916d35/devtools/server/actors/replay/react-devtools/contentScript.js#L22) to see where the backend component tree updates are being sent wrt the react components actually having been updated?

**Comment by bhackett1024** [comment]
Per https://github.com/RecordReplay/backend/issues/2051 there is also a delay installing the __RECORD_REPLAY_REACT_DEVTOOLS_SEND_MESSAGE__ hook.

**Comment by hbenl** [comment]
As suggested above, I added 
```javascript
console.log(`Annotation ${event} ${Date.now()}`);
```

[here](https://github.com/RecordReplay/gecko-dev/blob/5c85e5c35ce23cf0debaac26542ce82d43916d35/devtools/server/actors/replay/react-devtools/contentScript.js#L22).
Then I created a small React app with the following main component:
```typescript
export default function App() {
  const [list, setList] = useState<string[]>([]);

  useEffect(() => {
    console.log(`Rendered ${Date.now()}`);
  });

  const addItem = (newValue: string) => {
    setList([...list, newValue]);
    console.log(`Adding ${Date.now()}`);
  };

  const removeItem = (index: number) => {
    const newList = [...list];
    newList.splice(index, 1);
    setList(newList);
    console.log(`Removing ${Date.now()}`);
  }

  console.log(`Rendering ${Date.now()}`);

  return (
    <div className="App">
      <List list={list} removeItem={removeItem} />
      <Adder addItem={addItem} />
    </div>
  );
}
```
and made a recording of it where I add and then remove one item.
Here are the console messages I got in that recording, each message prefixed with its `executionPointTime`:
```
815 - Rendering 1628688121809
1255 - Rendered 1628688122249
1332 - Annotation isBackendStorageAPISupported 1628688122326
1333 - Annotation isSynchronousXHRSupported 1628688122327
1334 - Annotation operations 1628688122328
1336 - Annotation extensionBackendInitialized 1628688122330
5732 - Adding 1628688126727
5738 - Rendering 1628688126733
5884 - Rendered 1628688126879
5886 - Annotation operations 1628688126880
8131 - Removing 1628688129126
8133 - Rendering 1628688129128
8267 - Rendered 1628688129262
8272 - Annotation operations 1628688129266
```

And here are the times and points recorded in the annotations themselves:
```
isBackendStorageAPISupported - 1332 , 1622592768298983140574649380765722
isSynchronousXHRSupported - 1333 , 1622592768298985446417658594459676
operations - 1335 , 1622592768298987752260667808153630
extensionBackendInitialized - 1336 , 1622592768298990058103677021847584
operations - 5886 , 10060075163431283599850472741011589
operations - 8272 , 14927853468322805067130731847745780
```

So the ReactDevTools backend sends the annotations only a few milliseconds after changing the DOM and running our effect and the times recorded in the messages and annotations show that there is no further delay recording them.
The main delay (more than 100 milliseconds) is between the rendering call and the call to our effect. I assume that this time is spent updating the DOM.

In other words: my impression that we're getting the messages with a delay was wrong, I simply underestimated the time it can take in React for a state change to lead to a DOM change.
